File: app/src/main/java/exceptionapi/ex01.java
package exceptionapi;

public class ex01 {
    static int divide(int a, int b) {
        return a / b;
    }

    public static void main(String[] args) {
        try {
            System.out.println(divide(10, 0));
        } catch (ArithmeticException e) {
            System.out.println("error: " + e.getMessage());
        } finally {
            System.out.println("finally block executed");
        }
    }
}

File: app-kotlin/src/main/kotlin/exceptionapi/ex01.kt
package exceptionapi

class ex01 {
    companion object {
        fun divide(a: Int, b: Int): Int = a / b

        @JvmStatic
        fun main(args: Array<String>) {
            try {
                println(divide(10, 0))
            } catch (e: ArithmeticException) {
                println("error: ${e.message}")
            } finally {
                println("finally block executed")
            }
        }
    }
}
